### PR TITLE
Add implementation for governance token template and token factory

### PR DESCRIPTION
**Pull request by JasperTimm**
This is intended to cover: 
#4 
#5 

Things to consder:

- Do we want to explicitly disable the allowance (approval) related functions inherited from ERC20? (Probably)
- Where are we storing the list of Tokens created? In the factory or a separate contract?
- Where is the mapping to the org/repo stored?

TO DO:

- [ ] Add appropriate events to track token creation
- [ ] implement `balanceOfAt(uint256 blocknumber)`

### CODE REVIEW

File: contracts/OctobayGovTokenFactory.sol [hunk 1]
```diff
@@ -0,0 +1,21 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.6.0;
+pragma experimental ABIEncoderV2;
+
+import './OctobayStorage.sol';
+import './OctobayGovToken.sol';
+
+contract OctobayGovTokenFactory is OctobayStorage {
+
+    event NewTokenEvent(string name, string symbol);
```
Comment (mktcode) [inline contracts/OctobayGovTokenFactory.sol]: This event should also contain the address. That way our subgraph does not need to do any extra work.

File: contracts/OctobayGovToken.sol [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    function transferFrom(address _sender, address _recipient, uint256 _amount) public override onlyOctobay returns (bool) {
+        _transfer(_sender, _recipient, _amount);
+        return true;
+    }
+
+    /// @notice Explicitly disabling transfers by individual owners 
+    function transfer(address recipient, uint256 amount) public override returns (bool) {
+        require(false, "Transfers are only allowed by the Octobay contract");
+    }
+}
```
Comment (mktcode) [inline contracts/OctobayGovToken.sol]: File should end with empty line. (Don't know who invented this convention or why but even GitHub suggestes it :D)
I still have to set up proper linting for this and other repos.

**Review by mktcode — CHANGES_REQUESTED** [review]
Almost perfect! Just the one thing with the event. The list of tokens created should also be held by the factory I'd say, alongside with the ID of the repo/org. I'm just wondering if we need to store the info whether it's a repo or an org token.... I think yes. So the create function would need an `nodeId` and `nodeType`.

And one more idea: Transfering a token from a repo to an org then could be helpful when a project grows. Since you were that quick... we can maybe consider this.

File: contracts/OctobayGovToken.sol (hunk 2 above)
Comment (mktcode) [reply contracts/OctobayGovToken.sol]: The other change is obviously the more important one. Don't mind this one. xD I was just wondering why even GitHub wants us to put an empty line at the end. Anyway....

**Comment by JasperTimm** [comment]
Ok, made a couple of changes:

- The new token event now includes the address of the newly created token.
- Stored the mapping of the org/repo path to the token address in the factory. I've left it as a string for now, as I'm not sure the factory needs to know whether it's an org or repo but we can obviously change it later.
- There's a function to update the org/repo (which I've named 'project' for now)

**Comment by JasperTimm** [comment]
This can be closed without merging now, as the PR in #9 includes this.

**Pull request closed without merging**